src/lib/swapper/swappers/OsmosisSwapper/OsmosisSwapper.ts

```diff
@@ -340,11 +352,11 @@ export class OsmosisSwapper implements Swapper<ChainId> {
        * asset from the Osmosis network to the buy asset network.
        */
 
-      const amount = await pollForAtomChannelBalance(sellAddress, osmoUrl)
       const transfer = {
         sender: sellAddress,
         receiver: receiveAddress,
-        amount,
+        // IBC transfers are cheap compared to actual swaps, so we can rely on the slow "swap" tx fee
+        amount: bnOrZero(buyAmountCryptoBaseUnit).minus(swapFeeData.slow.txFee).toString(),
```

Comment:
We do need to deduct fees, slow Tx fee is the closest we can get to what can actually be IBC transferred. Sure, introspecting the actual balance would give you accurate amounts... assuming your balance was previously empty, or else it would IBC transfer your entire balance.